include/DTR_mf.hpp

```diff
@@ -0,0 +1,385 @@
+     *
+     * @param dimension_time_file An output stream for writing dimension and time information.
+     * @param verbose A flag indicating whether to print verbose output (default is true).
+     */
+    DTRProblem(std::ofstream& dimension_time_file, bool verbose = true);
+
+    /**
+     * @brief Compute the solution of the DTR problem.
+     *
+     * This method executes the setup, right-hand side assembly, solution, and output steps
+     * for the DTR problem for a given number of times. The number of initial refinements
+     * determines the number of cells in the mesh at the first solution, and the number of
+     * executed cycles determines the number of additional refinements performed.
+     *
+     * @param n_initial_refinements The number of initial refinements to perform on the mesh.
+     * @param n_cycles The number of solutions to compute by adding a refinement at each iteration.
+     */
+    void run(unsigned int n_initial_refinements = 3, unsigned int n_cycles = 9);
+
+    /**
+     * @brief Compute the error of the solution using the given norm type.
+     *
+     * @param norm_type The norm type to use for computing the error.
+     * @return The computed error.
+     */
+    double compute_error(const VectorTools::NormType &norm_type) const;
+
+    /**
+     * @brief Get the number of global active cells in the triangulation.
+     *
+     * @return The number of global active cells.
+     */
+    unsigned int get_cells() const { return triangulation.n_global_active_cells(); }
+
+    /**
+     * @brief Get the number of degrees of freedom.
+     *
+     * @return The number of degrees of freedom.
+     */
+    unsigned int get_dofs() const { return dof_handler.n_dofs(); }
+
+    /**
+     * @brief Get the degree of the finite element.
+     *
+     * @return The degree of the finite element.
+     */
+    unsigned int get_fe_degree() const { return degree_finite_element; }
+
+  private:
+    /**
+     * @brief Set up the system for solving the DTR problem.
+     *
+     * This method initializes the triangulation, degree of freedom handler, constraints,
+     * system matrix, multigrid constraints, and multigrid matrices.
+     */
+    void setup_system();
+
+    /**
+     * @brief Assemble the right-hand side of the DTR problem.
+     *
+     * This method computes the right-hand side vector for the linear system.
+     */
+    void assemble_rhs();
+
+    /**
+     * @brief Solve the DTR problem using the Conjugate Gradient method.
+     *
+     * This method solves the linear system using the Conjugate Gradient solver.
+     */
+    void solve();
+
+    /**
+     * @brief Output the results of the DTR problem for the given cycle.
+     *
+     * @param cycle The cycle number for which to output the results.
+     */
+    void output_results(const unsigned int cycle) const;
+
+#ifdef DEAL_II_WITH_P4EST
+    parallel::distributed::Triangulation<dim> triangulation;
+#else
+    Triangulation<dim> triangulation;
+#endif
```

Comment:
Is there a specific reason why you support triangulations both with and without p4est? If so, have you tested both?